graal/graal_bindist.bzl

```diff
@@ -126,15 +141,41 @@ _graal_native_image_version_configs = {
             "17": {
                 "darwin-amd64": "80ac09d45f8822413b9f16297da60da196013bbcfbc4bc7721f1257885ebe063",
                 "linux-amd64": "df488a04b5405c6443c90e94710cd3bd2be9adcb3768f91429aa494168d52440",
-            }
+            },
         },
     },
+    "22.0.0.2": {
+        "urls": ["https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-{version}/native-image-installable-svm-java{java_version}-{platform}-{version}.jar"],
+        "sha": {
+            "11": {
+                "darwin-amd64": "03c27de6cce61ee8073e89252212457f3fbac2c0bc9bfa4acbff12176476c176",
+                "linux-aarch64": "51d41e890a5aabf8e7b9d4f4e0f88206ee70a261f7dbb0315d51770ab8f3009e",
+                "linux-amd64": "8504a3441f5b28b8fd625f676674a9216f082ae63a4e30d43930c80f9672e71d",
+            },
+            "17": {
+                "darwin-amd64": "007fa742cd139d447f83d776b6d78e717c9df11d56a61061a5937547c20028b7",
+                "linux-aarch64": "798947d0a93988929d2b8e3555f7c65225e789124cd99fbc0c3aae5f350175db",
+                "linux-amd64": "8c25f650d58c2649c97061cb806dfaec9e685d5d2b80afc7cf72fe61d6891831",
+            },
+        },
+    }
 }
 
 def _get_platform(ctx):
+    res = ctx.execute(["uname", "-p"])
+    arch = "amd64"
+    if res.return_code == 0:
+        uname = res.stdout.strip()
+        if uname == "arm":
+            arch = "arm64"
+        elif uname == "aarch64":
+            arch = "aarch64"
+
     if ctx.os.name == "linux":
-        return "linux-amd64"
+        return "linux-%s" % arch
     elif ctx.os.name == "mac os x":
+        if arch == "arm64" or arch == "aarch64":
+            print("GraalVM has no distribution yet for ARM-based macOS. Using `amd64` instead.")
```

Comment:
I wonder if this actually works on arm or if we should error in this case.

Reply:
it does for me at least, due to Rosetta, but least lmk if you encounter any issues 👍🏻